kaolin/transforms/transforms.py

```diff
@@ -36,7 +36,45 @@
 from kaolin.transforms import voxelfunc
 
 
-class Compose(object):
+class Transform(object):
+    """Base class for all Kaolin transforms.
+
+    This class generates __repr__ string automatically. Given that all attributes
+    have valid __repr__, it generates a string with the following format:
+    .. code-block::
+       MyClass(a=1, b=2, ...)
+
+    The attributes are sorted alphabetically to ensure determinism.
```

Comment:
I don't see where you are ordering alphabetically, I believe dict keys ordering depend on python version.

Reply:
I believe Python (starting from 3.7) only guarantees dict key to be sorted by insertion order; so I am not sure if that guarantees two objects with the same attributes will always get the same repr. Sorting alphabetically guarantees that :P

Reply:
Yes, but, where are you enforcing alphabetical order ?

Reply:
Yes, but, where are you enforcing alphabetical order ?

Reply:
https://github.com/NVIDIAGameWorks/kaolin/blob/8937233ba30059eb3d003d1add0fad1ee54c4c1a/kaolin/transforms/transforms.py#L69

Reply:
Missed that line :)